### DISCUSSION TITLE
How to configure complementary  PWM channels on STM32F4?

### DISCUSSION THREAD

**Discussion by gabopushups** [zephyrproject-rtos/zephyr#37655] [Q&A]
Is it possible  to configure complementary PWM channels on the STM32F4 using zephyr?

How do I declare, for instance, TIM1_CH1N (PE9)? So far, I've tried the following:

```
   &timers1 {
	status = "okay";
	pwm1: pwm {
		status = "okay";
		pinctrl-0 = <&tim1_ch1n_pe8
				   &tim1_ch1_pe9>;
	};
   };

```
But then, when I need to use the pwm bindings to declare them, I don't know how to tell that a channel should be negated.

```

     mypwm {
		compatible = "pwm-leds";
		pwm_motor: pwm_motor {
			pwms = <&pwm1 1 0 PWM_POLARITY_NORMAL>;
		};

```

How should I modify `pwms = <&pwm1 1 0 PWM_POLARITY_NORMAL>;` to use the complementary channel? Since the parameters only include the timer module that I declared earlier, the channel, period and flags, and that declaration would run the PWM on PE9

**Comment by FelixEL** [comment]
pwm1: pwm_1 {
	pwms = <&pwm1 1 PWM_MSEC(20) (PWM_POLARITY_NORMAL|**STM32_PWM_COMPLEMENTARY**)>;
};

https://docs.zephyrproject.org/latest/build/dts/api/bindings/pwm/st%2Cstm32-pwm.html
